internal/armada/server/submit.go

```diff
@@ -897,8 +897,23 @@ func (server *SubmitServer) applyDefaultsToPodSpec(spec *v1.PodSpec) {
 	if shouldDefaultPriorityClass {
 		spec.PriorityClassName = defaultPriorityClass
 	}
+
+	// add missing TerminationGracePeriod if needed
+	server.applyTerminationGracePeriodDefault(spec)
 }
 
+// applyTerminationGracePeriodDefault will give the podspec a default if needed
+func (server *SubmitServer) applyTerminationGracePeriodDefault(spec *v1.PodSpec) {
+	terminationGracePeriod := int64(server.schedulingConfig.TerminationGracePeriod.Default.Seconds())
+	preememptionEnabled := server.schedulingConfig.Preemption.Enabled
+	if preememptionEnabled &&
+		spec.PriorityClassName != "" &&
+		spec.TerminationGracePeriodSeconds == nil &&
```

Comment:
I'd extract the conditions into named variables so it is clearer, eg. this could be `terminationGracePeriodSecondsNotDefined`